pkg/model/components/apiserver.go

```diff
@@ -133,8 +133,6 @@ func (b *KubeAPIServerOptionsBuilder) BuildOptions(o interface{}) error {
 
 	c.AllowPrivileged = fi.Bool(true)
 	c.ServiceClusterIPRange = clusterSpec.ServiceClusterIPRange
-	c.EtcdServers = []string{"http://127.0.0.1:4001"}
-	c.EtcdServersOverrides = []string{"/events#http://127.0.0.1:4002"}
 
```

Comment:
Trying to explain my previous comment, model section:
```suggestion

	for _, etcdCluster := range clusterSpec.EtcdClusters {
		switch etcdCluster.Name {
		case "main":
			if etcdCluster.EnableEtcdTLS {
				c.EtcdServers = append(c.EtcdServers, "https://127.0.0.1:4001")
			} else {
				c.EtcdServers = append(c.EtcdServers, "http://127.0.0.1:4001")
			}
		case "events":
			if etcdCluster.EnableEtcdTLS {
				c.EtcdServersOverrides = append(c.EtcdServersOverrides, "/events#https://127.0.0.1:4002")
			} else {
				c.EtcdServersOverrides = append(c.EtcdServersOverrides, "/events#http://127.0.0.1:4002")
			}
		}
	}

```